domain.go

```diff
@@ -167,28 +148,33 @@ func getCaseSensitivitySourceAttributeList(attributeMap []AttributeMap) map[stri
 // (Create, Update and Delete) based on whether they are in the slice
 //  of destination Person instances.
 // It skips all source Person instances that have DisableChanges set to true
-func GenerateChangeSet(sourcePeople, destinationPeople []Person, attributeMap []AttributeMap) ChangeSet {
+func GenerateChangeSet(sourcePeople, destinationPeople []Person, attributeMap []AttributeMap, idField string) ChangeSet {
 	var changeSet ChangeSet
 
-	// Find users who need to be created
+	// Find users who need to be created or updated
 	for _, sp := range sourcePeople {
 		// If user was missing a required attribute, don't change their record
 		if sp.DisableChanges {
 			continue
 		}
 
-		personInDestinationStatus := PersonStatusInList(sp, destinationPeople, attributeMap)
-		switch personInDestinationStatus {
-		case PersonIsNotInList:
+		destinationPerson := getPersonFromList(sp.CompareValue, destinationPeople)
+		if destinationPerson.CompareValue == "" {
 			changeSet.Create = append(changeSet.Create, sp)
-		case PersonIsInListButDifferent:
+			continue
+		}
+
+		if !personAttributesAreEqual(sp, destinationPerson, attributeMap) {
+			sp.ID = destinationPerson.Attributes["id"]
 			changeSet.Update = append(changeSet.Update, sp)
+			continue
```

Comment:
correct, I put it in just to be explicit and in case any code was added to that function later below that check we'd have to consider the impact and whether to remove the continue or not